operations/host_spawn.go

```diff
@@ -1375,6 +1375,62 @@ Examples:
 	}
 }
 
+func hostFindBy() cli.Command {
+	const (
+		ipAddressFlagName = "ip-address"
+		legacyFlagName    = "legacy"
+	)
+	return cli.Command{
+		Name:  "get-host",
+		Usage: "look up existing hosts",
+		Flags: addHostFlag(
+			cli.StringFlag{
+				Name:  joinFlagNames(ipAddressFlagName, "ip"),
+				Usage: "ip address used to find host",
+			},
+			cli.BoolFlag{
+				Name:  legacyFlagName,
+				Usage: "returns host link in spruce",
+			},
+		),
+		Before: requireAtLeastOneFlag(ipAddressFlagName),
+		Action: func(c *cli.Context) error {
+			confPath := c.Parent().Parent().String(confFlagName)
+			ipAddress := c.String(ipAddressFlagName)
+			legacy := c.Bool(legacyFlagName)
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			conf, err := NewClientSettings(confPath)
+			if err != nil {
+				return errors.Wrap(err, "problem loading configuration")
+			}
+			client := conf.getRestCommunicator(ctx)
+			defer client.Close()
+
+			host, err := client.FindHostByIpAddress(ctx, ipAddress)
+			if err != nil {
+				return errors.Wrapf(err, "failed to fetch host with ip address '%s'", ipAddress)
+			}
+			if host == nil {
+				return errors.Errorf("Host with ip address '%s' not found", ipAddress)
+			}
+
+			link := fmt.Sprintf("https://spruce.mongodb.com/host/%s", utility.FromStringPtr(host.Id))
+			if legacy {
+				link = fmt.Sprintf("https://evergreen.mongodb.com/host/%s", utility.FromStringPtr(host.Id))
```

Comment:
Actually sorry I didn't catch this earlier; we really should be using conf.UIServerHost. If we don't do this, it doesn't work for staging / local, and theoretically legacy / spruce should be handled from there (and if they want to switch to legacy they can do it from the UI). That being said, we can probably drop the legacy flag altogether.

Reply:
done